Tests/RxSwiftTests/Tests/Observable+StandardSequenceOperatorsTest.swift

```diff
@@ -3181,6 +3181,598 @@ extension ObservableStandardSequenceOperatorsTest {
     
 }
 
+
+// MARK: concatMap
+extension ObservableStandardSequenceOperatorsTest {
+    
+    func testConcatMap_Complete() {
+        let scheduler = TestScheduler(initialClock: 0)
+        
+        let xs = scheduler.createHotObservable([
+            next(5, scheduler.createColdObservable([
+                error(1, testError)
+                ])),
+            next(105, scheduler.createColdObservable([
+                error(1, testError)
+                ])),
+            next(300, scheduler.createColdObservable([
+                next(10, 102),
+                next(90, 103),
+                next(110, 104),
+                next(190, 105),
+                next(440, 106),
+                completed(460)
+                ])),
+            next(400, scheduler.createColdObservable([
+                next(180, 202),
+                next(190, 203),
+                completed(205)
+                ])),
+            next(550, scheduler.createColdObservable([
+                next(10, 301),
+                next(50, 302),
+                next(70, 303),
+                next(260, 304),
+                next(310, 305),
+                next(1000, 306),
+                completed(1050)
+                ])),
+            next(750, scheduler.createColdObservable([
+                completed(40)
+                ])),
+            next(850, scheduler.createColdObservable([
+                next(80, 401),
+                next(90, 402),
+                completed(100)
+                ])),
+            completed(900)
+            ])
+        
+        let res = scheduler.start(10000) {
+            xs.concatMap { $0 }
+        }
+        
+        
+        XCTAssertEqual(res.events, [
+            next(310, 102),
+            next(390, 103),
+            next(410, 104),
+            next(490, 105),
+            next(740, 106),
+            next(940, 202),
+            next(950, 203),
+            next(975, 301),
+            next(1015, 302),
+            next(1035, 303),
+            next(1225, 304),
+            next(1275, 305),
+            next(1965, 306),
+            next(2135, 401),
+            next(2145, 402),
+            completed(2155),
+            ]
+        )
+        
+        XCTAssertEqual(xs.subscriptions, [
+            Subscription(200, 900)
+            ])
+        
+        
+        let innerSubscriptions = xs.recordedEvents[2..<xs.recordedEvents.endIndex].flatMap { $0.value.element?.subscriptions ?? [] }
```

Comment:
I'm not sure why do we start here from 2, it looks like we could start from 0, but maybe comment below is more important ....

Comparing a list of flat mapped subscription results could theoretically be different then expected, and still produce a valid result. I know this is a exponentially unlikely long shot, but my OCD is alarmed :), and would make unit test code more idiomatic.

Maybe it would be better to assert each subscription manually. I understand that it's little more code, but it's more exact IMHO ¯\_(ツ)_/¯